bubblegum/program/src/tests/simple.rs

```diff
@@ -0,0 +1,95 @@
+use solana_program_test::tokio;
+use solana_sdk::signature::{Keypair, Signer};
+
+use super::{
+    clone_keypair,
+    context::{BubblegumTestContext, DEFAULT_LAMPORTS_FUND_AMOUNT},
+    LeafArgs, Result, Tree,
+};
+
+// Test for multiple combinations?
+const MAX_DEPTH: usize = 20;
+const MAX_BUF_SIZE: usize = 64;
+
+// TODO: test signer conditions on mint_authority and other stuff that's manually checked
+// and not by anchor (what else is there?)
+
+// TODO: will add some exta checks to the tests below (i.e. read accounts and
+// assert on values therein).
+
+// Creates a `BubblegumTestContext`, a `Tree` with default arguments, and also mints an NFT
+// with the default `LeafArgs`.
+async fn context_tree_and_leaf() -> Result<(
+    BubblegumTestContext,
+    Tree<MAX_DEPTH, MAX_BUF_SIZE>,
+    LeafArgs,
+)> {
+    let context = BubblegumTestContext::new().await?;
+
+    let (tree, leaf) = context
+        .default_create_and_mint::<MAX_DEPTH, MAX_BUF_SIZE>()
+        .await?;
+
+    Ok((context, tree, leaf))
+}
+
+#[tokio::test]
+async fn test_create_tree_and_mint_passes() {
+    context_tree_and_leaf().await.unwrap();
+}
+
+#[tokio::test]
+async fn test_creator_verify_and_unverify_passes() {
+    let (context, tree, mut leaf) = context_tree_and_leaf().await.unwrap();
+
+    tree.verify_creator(&leaf, &context.default_creators[0])
+        .await
+        .unwrap();
+    // Reflect change.
+    leaf.metadata.creators[0].verified = true;
+
+    tree.unverify_creator(&leaf, &context.default_creators[0])
+        .await
+        .unwrap();
+    leaf.metadata.creators[0].verified = false;
+}
+
+#[tokio::test]
+async fn test_delegate_passes() {
+    let (_, tree, mut leaf) = context_tree_and_leaf().await.unwrap();
+    let new_delegate = Keypair::new();
+
+    tree.delegate(&leaf, new_delegate.pubkey()).await.unwrap();
+    // Reflect changes.
+    leaf.delegate = new_delegate;
```

Comment:
So for tests like this where you don't use the local state again, is this update to reflect the change just being done for posterity like if someone were to use this as a basis for a more advanced test, or add more things afterwards?  My assumption is that `leaf` is dropped at the end of this test.

Reply:
Yes, within the scope of this PR it's for illustrative purposes, but there are some more testing checks coming afterwards where we're actually attempting operations that use/need the new values.

Reply:
ok